### PR TITLE
[AIRFLOW-3649] Feature to add extra labels to kubernetes worker pods

### PR DESCRIPTION
**Pull request by wyndhblb**
Make sure you have checked _all_ steps below.

### Jira

- [x] https://issues.apache.org/jira/browse/AIRFLOW-3649

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

This pr is meant to add the ability to attached a static set of extra labels to worker pods for resource, metrics, and other use tracking.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by wyndhblb** [comment]
unclear why ```test_trigger_dag_button (tests.www_rbac.test_views.TestTriggerDag) ... FAILED``` fails on py3, but i don't think it's related to this PR

**Comment by feng-tao** [comment]
once you rebase with master, it should be good @wyndhblb

**Comment by feng-tao** [comment]
PTAL @dimberman

File: airflow/contrib/executors/kubernetes_executor.py [hunk 1]
```diff
@@ -252,6 +265,15 @@ def _validate(self):
                 'or `dags_in_image` '
                 'or `git_repo and git_branch and git_dags_folder_mount_point`')
 
+        # cannot allow certain labels, as airflow needs them
+        if self.kube_extra_labels:
+            for k in self.protected_labels:
+                if k in self.kube_extra_labels:
+                    raise AirflowConfigException(
+                        'In kubernetes mode `extra_labels` cannot include '
+                        '`airflow-worker`, `dag_id`, `task_id`, `try_number`, or `execution_date`'
```
Comment (andriisoldatenko) [inline airflow/contrib/executors/kubernetes_executor.py]: I prefer to use `self.protected_labels` instead of hardcode it here ^

Comment (dimberman) [reply airflow/contrib/executors/kubernetes_executor.py]: I agree with this.

**Review by dimberman — CHANGES_REQUESTED** [review]
Overall LGTM, just address @andriisoldatenko 's comment and would be glad to merge

**Comment by andriisoldatenko** [comment]
@wyndhblb do you need any help or may be I can finish this PR?
cc @ashb @dimberman

**Comment by wyndhblb** [comment]
@andriisoldatenko hi didn't see this till my name was pinged .. i've updated things

**Comment by ashb** [comment]
Closing this in favour of the slightly simpler #5134.

**Pull request closed without merging**